parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java

```diff
@@ -136,6 +154,19 @@ public static Metrics footerMetrics(ParquetMetadata metadata, MetricsConfig metr
         toBufferMap(fileSchema, lowerBounds), toBufferMap(fileSchema, upperBounds));
   }
 
+  private static MessageType getParquetType(ParquetMetadata metadata, Schema expectedSchema, NameMapping nameMapping) {
```

Comment:
The idea here is that we should import metrics only for columns in the current schema.